src/EditorFeatures/Core/InlineDiagnostics/InlineDiagnosticsOptions.cs

```diff
@@ -11,32 +11,29 @@
 
 namespace Microsoft.CodeAnalysis.Editor.InlineDiagnostics
 {
-    internal static class InlineDiagnosticsOptions
+    [ExportGlobalOptionProvider, Shared]
+    internal sealed class InlineDiagnosticsOptions : IOptionProvider
     {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineDiagnosticsOptions()
+        {
+        }
+
+        ImmutableArray<IOption> IOptionProvider.Options { get; } = ImmutableArray.Create<IOption>(
+            EnableInlineDiagnostics,
+            Location);
+
         public static readonly PerLanguageOption2<bool> EnableInlineDiagnostics =
-            new(nameof(InlineDiagnosticsOptions),
-                nameof(EnableInlineDiagnostics),
+            new("InlineDiagnosticsOptions",
```

Comment:
Why is not good to use `nameof()` here? Easier to move options between code files?

Reply:
Because renaming the internal type might break the option (change the key that's used to look it up). There is really no relationship that we need to maintain between the type that declares the option and the option key.